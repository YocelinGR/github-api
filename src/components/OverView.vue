<template>
  <div>
    <section v-for="repo in user.repositories && user.repositories.nodes" :key="repo.id">
      <a :href="repo.url" class="title">{{repo.name}}</a>
      <p v-if="repo.repositoryTopics.nodes">
        <span v-for="topic in repo.repositoryTopics.nodes" :key="topic.id">
          {{topic.topic.name}}
        </span>
      </p>
    </section>
  </div>
</template>

<script>
import gql from 'graphql-tag'

const overviewTag = gql`
  query user {
  user(login: "YocelinGR"){
    id
    repositories(first: 6){
      nodes{
        id
        name
        url
        repositoryTopics(first: 3){
          nodes{
            id
            resourcePath
            url
            topic{
              id
              name
            }
          }
        }
      }
    }
  }
}
`;
export default {
  data: () => ({
    user: {}
  }),
  apollo: {
    user: {
      query: overviewTag
    }
  }
}
</script>

<style scoped>
  section {
    border: 1px rgb(141, 134, 134) solid;
    width: 45%;
    height: 100px;;
    display: inline-block;
    justify-content: center;
    align-items: center;
    margin: 15px;
    padding: 15px;
    font-size: 16px;
    color: rgb(64, 103, 212);
    font-weight: 600;
    border-radius: 5px;
  }
  span {
    color: rgb(100, 97, 97);
    font-weight: 600;
    padding: 8px;
  }
  .title {
    font-size: 16px;
    color: rgb(10, 112, 196);
    padding: 4px;
  }
</style>
