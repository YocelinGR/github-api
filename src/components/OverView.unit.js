import OverView from './OverView.vue'
import { createLocalVue } from '@vue/test-utils'

describe('@components/OverView', () => {
  const localVue = createLocalVue()
  it('OverView is a vue instance', () => {
    const wrapper = shallowMount(OverView, { localVue })
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('Displays overview first userÂ´s repos', () => {
    const wrapper = shallowMount(OverView, { localVue })
    wrapper.setData({
      user: {
        id: "MDQ6VXNlcjM5ODMzMDQ1",
        repositories: {
          nodes: [
            {
              id: "MDEwOlJlcG9zaXRvcnkxMzU4NDM4NzQ=",
              name: "TriviaPreAdmision",
              url: "https://github.com/YocelinGR/TriviaPreAdmision",
              repositoryTopics: {
                nodes: []
              }
            }
          ]
        }
      },
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
