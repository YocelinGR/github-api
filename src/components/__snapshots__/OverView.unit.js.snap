// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@components/OverView Displays overview first userÂ´s repos 1`] = `
<div>
  <section>
    <a
      class="title"
      href="https://github.com/YocelinGR/TriviaPreAdmision"
    >
      TriviaPreAdmision
    </a>
     
    <p />
  </section>
</div>
`;
